WPFHospitalEditor/EquipmentRelocation.xaml.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows;
+using WPFHospitalEditor.Controller;
+using WPFHospitalEditor.Controller.Interface;
+using WPFHospitalEditor.DTOs;
+using WPFHospitalEditor.Model;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for EquipmentRelocation.xaml
+    /// </summary>
+    public partial class EquipmentRelocation : Window
+    {
+        private IRoomServerController roomServerController = new RoomServerController();
+        private IEquipmentServerController equipmentServerController = new EquipmentServerController();
+        private IExaminationServerController examinationServerController = new ExaminationServerController();
+        public ObservableCollection<RoomsWithChosenEquipmentAmount> roomsWithEquipmenAmount { get; set; }
+        private HospitalMap hospitalMap;
+        private int destinationRoomId;
+        private string relocationEquipmentName;
+        private DateTime startDate;
+        private DateTime endDate;
+
+        public EquipmentRelocation(string relocationEquipmentName, int roomId, HospitalMap hospitalMap)
+        {
+            SearchResultDialog.selectedObjectId = -1;
+            InitializeComponent();
+            equipmentName.Text += relocationEquipmentName;
+            this.destinationRoomId = roomId;
+            this.relocationEquipmentName = relocationEquipmentName;
+            this.hospitalMap = hospitalMap;
+            roomsWithEquipmenAmount = new ObservableCollection<RoomsWithChosenEquipmentAmount>();
+            GetRoomsWithGivenEquipmentAmount();
```

Comment:
You can avoid calling two methods, and immediately call FillObservableCollection where you can initialize obser. collection as well.

Reply:
Fixed.